updater/src/config.rs

```diff
@@ -0,0 +1,139 @@
+                workflow_url: "https://api.github.com/repos/obhq/obliteration/actions/workflows/36859008/runs".to_string(),
+
+                game_images_folder: "./images/games/".to_string(),
+                homebrew_images_folder: "./images/hb/".to_string(),
+                homebrew_database: "./HBstore.db".to_string(),
+                game_skips_database: "./game_skips.json".to_string(),
+                database: "./database.json".to_string(),
+                stats_file: "./stats.json".to_string(),
+
+                tag_homebrew: 6164722453,
+                tag_playable: 6164497050,
+                tag_ingame: 6164500133,
+                tag_intro: 6164505028,
+                tag_boots: 6164509950,
+                tag_nothing: 6164514963,
+            };
+
+            // create default config
+            fs::create_dir_all(FOLDER_CONFIG)?;
+            fs::write(&config_file, toml::to_string_pretty(&config)?)?;
+
+            config
+        } else {
+            let toml_content = fs::read_to_string(&config_file)?;
+            toml::from_str(&toml_content)?
+        }
+    };
+
+    // create needed folders
+    let game_images_folder = Path::new(&config_data.game_images_folder);
+    let homebrew_images_folder = Path::new(&config_data.homebrew_images_folder);
+
+    let database_folder = Path::new(&config_data.database).parent().unwrap_or_else(|| {
+        panic_red!("Error while getting the parent folder for: \"{}\"", &config_data.database);
+    });
+
+    let hb_database_folder = Path::new(&config_data.homebrew_database).parent().unwrap_or_else(|| {
+        panic_red!("Error while getting the parent folder for: \"{}\"", &config_data.database);
+    });
+
+    let game_skips_database_folder = Path::new(&config_data.game_skips_database).parent().unwrap_or_else(|| {
+        panic_red!("Error while getting the parent folder for: \"{}\"", &config_data.database);
+    });
+
+    let stats_file_folder = Path::new(&config_data.stats_file).parent().unwrap_or_else(|| {
+        panic_red!("Error while getting the parent folder for: \"{}\"", &config_data.database);
+    });
+
+
+    if !game_images_folder.exists() {
+        fs::create_dir_all(game_images_folder)?;
+    }
+
+    if !homebrew_images_folder.exists() {
+        fs::create_dir_all(homebrew_images_folder)?;
+    }
+
+    if !database_folder.exists() {
+        fs::create_dir_all(database_folder)?;
+    }
+
+    if !hb_database_folder.exists() {
+        fs::create_dir_all(hb_database_folder)?;
+    }
+
+    if !game_skips_database_folder.exists() {
+        fs::create_dir_all(game_skips_database_folder)?;
+    }
+
+    if !stats_file_folder.exists() {
+        fs::create_dir_all(stats_file_folder)?;
+    }
+
+    // get environment values :3
+    let secrets = Secrets {
+        github_api_token: match env::var("GITHUB_API_TOKEN") {
```

Comment:
IIRC we can query the public repository without the need of API token.

Reply:
yes we can, however there is a limit of how many requests you can do per hour. (Also github actions gives this api token automatically, so we don't even need to use our own).